src/indigo-react/lib/useForm.js

```diff
@@ -192,11 +192,16 @@ export default function useForm(inputConfigs = []) {
       configs.map(
         ({ name, error, autoFocus, disabled, initialValue, ...rest }, i) => {
           const value =
-            values[name] || (initialValue === undefined ? '' : initialValue);
+            datas[i] !== undefined
+              ? datas[i]
+              : initialValue !== undefined
+              ? initialValue
+              : '';
           return {
             // Input props
             name,
-            data: datas[i],
+            value: value,
+            data: passes[i] ? datas[i] : undefined,
```

Comment:
now `value` is the raw value and `data` is either `value` or `undefined` depending on pass state

this lets us use `data` in callers and assume that if it's not undefined it's valid

Reply:
right yeah — lets us remove the onValue property within Input which was pretty hacky